barretenberg/cpp/pil/vm2/data_copy.pil

```diff
@@ -105,132 +133,147 @@ include "context.pil";
+    //////////////////////////////////////
+    // Computing the src index upper bound
+    //////////////////////////////////////
     // Computing the read count, i.e. the number of elements that will be read from the src data.
-    // We compute the max read index using min(offset + copy_size, src_data_size)
+    // We compute the data index upper bound using min(offset + copy_size, src_data_size)
     // This ensures that we cannot read pass the designated data address assigned by the parent/child
     // The min operation is essentially checking the comparison of the following
     // 1) (offset + copy_size) > src_data_size or
     // 2) (offset + copy_size) <= src_data_size
-    // if (1) then max_read_index = src_data_size, otherwise max_read_index = (offset + copy_size)
+    // if (1) then data_index_upper_bound = src_data_size, otherwise data_index_upper_bound = (offset + copy_size)
     pol commit offset_plus_size;
     offset_plus_size = sel_start * (offset + copy_size);
-    pol commit offset_plus_size_is_gt;
+    pol commit offset_plus_size_is_gt; // @boolean follows from the lookup into gt.
 
-    #[MAX_READ_INDEX_GT]
+    #[OFFSET_PLUS_SIZE_IS_GT_DATA_SIZE]
     sel_start { offset_plus_size, src_data_size, offset_plus_size_is_gt }
     in
     gt.sel_others { gt.input_a, gt.input_b, gt.res };
 
-    // Set max_read_index based on the conditions (1) or (2) from above
-    pol commit max_read_index;
-    max_read_index = sel_start * ((src_data_size - offset_plus_size) * offset_plus_size_is_gt + offset_plus_size);
+    // Set data_index_upper_bound based on the conditions (1) or (2) from above
+    pol commit data_index_upper_bound;
+    data_index_upper_bound = sel_start * ((src_data_size - offset_plus_size) * offset_plus_size_is_gt + offset_plus_size);
 
     //////////////////////////////
     // Error Handling
     //////////////////////////////
-    // Both errors are constrained to be boolean by the lookup into gt. (provided that sel_start == 1).
-    pol commit src_out_of_range_err; // Read slices should be in MEM range
-    pol commit dst_out_of_range_err; // Write slices should be in MEM range
-
-    pol commit max_mem_addr; // todo: While we do not support constants
-    sel_start * (max_mem_addr - constants.AVM_HIGHEST_MEM_ADDRESS) = 0;
-
-    // MAX_READ_ADDR = 0 if this is a top level call since there aren't any memory reads at the top level
-    // conceptually cd copy and rd copy don't perform reads at the top level.
-    pol MAX_READ_ADDR = src_addr + max_read_index;
-    pol commit max_read_addr;
-    max_read_addr = sel_start * MAX_READ_ADDR;
+    // Errors on whether the read or write addresses are out of range in memory. (provided that sel_start == 1).
+    pol commit src_out_of_range_err; // @boolean follows from the lookup into gt.
+    pol commit dst_out_of_range_err; // @boolean follows from the lookup into gt.
+
+    pol commit mem_size; // todo: While we do not support constants
```

Comment:
feels strange to rename this - is this the new naming convention we are using for the max accessible address in the other files as well?